### PR TITLE
DAOS-13530 test: Add a 'Functional Hardware Medium MD on SSD' stage

### PR DESCRIPTION
**Pull request by phender**
Enable running tests with MD on SSD as part of PRs and daily builds.

Skip-unit-tests: true
Skip-fault-injection-test: true
Allow-unstable-test: true
Test-tag: test_setup_vm test_setup

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: Jenkinsfile
Comment (brianjmurrell) [inline Jenkinsfile:199]: We don't `camelCase` in `Jenkinsfile`.

File: Jenkinsfile
Comment (brianjmurrell) [inline Jenkinsfile:1254]: I'm not aware that you can mix scripted and declarative in this way.  I have never come across it before, but I have never gone in search of it either..

File: Jenkinsfile
Comment (brianjmurrell) [inline Jenkinsfile:212]: Ditto.

File: Jenkinsfile
Comment (phender) [reply Jenkinsfile]: I found it in https://stackoverflow.com/questions/42837066/can-i-create-dynamically-stages-in-a-jenkins-pipeline and was able to verify it in debug build https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12986/26/, but it was a very simple scripted stage.  Given that we need to access parameters defined within the pipeline, I believe I need to define the map for parallel directly within the pipeline definition and use a groovy script to return the scripted stage.

File: Jenkinsfile
Comment (daltonbohning) [inline Jenkinsfile:294]: SDD -> SSD
```suggestion
        string(name: 'FUNCTIONAL_HARDWARE_MEDIUM_MD_ON_SSD_LABEL',
```

Comment (phender) [reply Jenkinsfile]: Thanks.  It was misspelled in its definition too.

**Comment by phender** [comment]
Blocked by https://github.com/daos-stack/pipeline-lib/pull/393

**Comment by phender** [comment]
All stages, except HW Medium UCX (due to this being a PR), ran in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-12986/98/pipeline/1228/

**Comment by phender** [comment]
Running HW Medium UCX stage manually w/ build params in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12986/99/

**Review by daltonbohning — CHANGES_REQUESTED** [review]
LGTM but -1 just for pipeline lib reference

**Comment by phender** [comment]
All hardware stages passed in manual (build with parameters) build https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12986/113

**Comment by phender** [comment]
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12986/114/ completed with known failures: https://daosio.atlassian.net/browse/DAOS-14554

**Comment by phender** [comment]
> LGTM but -1 just for pipeline lib reference

The pipeline-lib reference has been removed, but https://github.com/daos-stack/pipeline-lib/pull/393 will still need to land before this pr can land.

**Review by brianjmurrell — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Comment by phender** [comment]
The failures in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12986/117/testReport/ are known issues:
- https://daosio.atlassian.net/browse/DAOS-14227
- https://daosio.atlassian.net/browse/DAOS-14590
- https://daosio.atlassian.net/browse/DAOS-14554

**Comment by phender** [comment]
https://daosio.atlassian.net/browse/DAOS-14554 probably blocks landing this as it will cause new PRs to fail with the new `Functional Hardware Medium MD on SSD` stage.

**Comment by phender** [comment]
> https://daosio.atlassian.net/browse/DAOS-14554 probably blocks this PR from landing as it will cause new PRs to fail in the new `Functional Hardware Medium MD on SSD` stage.

I've added skip list entries to skip any of the failing tests in DAOS-14554 when they are run in the new `Functional Hardware Medium ND on SSD` stage:
```
/scratch/CI-skip-list-master:[['DAOS-14554', 'stage_name', 'Functional Hardware Medium MD on SSD', 'test_method_name', 'test_aggregation_punching']]
/scratch/CI-skip-list-master:[['DAOS-14554', 'stage_name', 'Functional Hardware Medium MD on SSD', 'test_method_name', 'test_daos_vol_mpich']]
/scratch/CI-skip-list-master:[['DAOS-14554', 'stage_name', 'Functional Hardware Medium MD on SSD', 'test_method_name', 'test_ioconsistency']]
/scratch/CI-skip-list-master:[['DAOS-14554', 'stage_name', 'Functional Hardware Medium MD on SSD', 'test_method_name', 'test_smallfilecount']]
/scratch/CI-skip-list-master:[['DAOS-14554', 'stage_name', 'Functional Hardware Medium MD on SSD', 'test_method_name', 'test_ior_small']]
/scratch/CI-skip-list-master:[['DAOS-14554', 'stage_name', 'Functional Hardware Medium MD on SSD', 'test_method_name', 'test_mdtest_small']]
/scratch/CI-skip-list-master:[['DAOS-14554', 'stage_name', 'Functional Hardware Medium MD on SSD', 'test_method_name', 'test_romio']]

```

**Pull request merged**
